### ISSUE TITLE
Specific URL raises error from chardet

### ISSUE THREAD

**Issue by Vexs** [JWCook/aiohttp-client-cache#75]
### The problem
When attempting to retrieve a result from the following URL with the following snippet:
```py
session = CachedSession(cache=SQLiteBackend('demo_cache'))
url = "https://news.search.yahoo.com/search;_ylt=AwrXnCKM_wFfoTAA8HbQtDMD;_ylu=X3oDMTEza3NiY3RnBGNvbG8DZ3ExBHBvcwMxBHZ0aWQDBHNlYwNwYWdpbmF0aW9u?p=test&nojs=1&ei=UTF-8&b=21&pz=10&bct=0&xargs=0"
await session.get(url)
```
The following error raises:
```py
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-20-5ba3f9a63552> in <module>
----> 1 await session.get(url)

c:\users\james\appdata\local\programs\python\python38\lib\site-packages\forge\_revision.py in inner(*args, **kwargs)
    320                 # pylint: disable=E1102, not-callable
    321                 mapped = inner.__mapper__(*args, **kwargs)
--> 322                 return await callable(*mapped.args, **mapped.kwargs)
    323         else:
    324             @functools.wraps(callable)  # type: ignore

c:\users\james\appdata\local\programs\python\python38\lib\site-packages\aiohttp_client_cache\session.py in _request(self, method, str_or_url, **kwargs)
     33             new_response = await super()._request(method, str_or_url, **kwargs)  # type: ignore
     34             await new_response.read()
---> 35             await self.cache.save_response(new_response, actions)
     36             return set_response_defaults(new_response)
     37 

c:\users\james\appdata\local\programs\python\python38\lib\site-packages\aiohttp_client_cache\backends\base.py in save_response(self, response, actions)
    158 
    159         logger.debug(f'Saving response for key: {actions.key}')
--> 160         cached_response = await CachedResponse.from_client_response(response, actions.expires)
    161         await self.responses.write(actions.key, cached_response)
    162 

c:\users\james\appdata\local\programs\python\python38\lib\site-packages\aiohttp_client_cache\response.py in from_client_response(cls, client_response, expires)
     89         if client_response.history:
     90             response.history = (
---> 91                 *[await cls.from_client_response(r) for r in client_response.history],
     92             )
     93         return response

c:\users\james\appdata\local\programs\python\python38\lib\site-packages\aiohttp_client_cache\response.py in <listcomp>(.0)
     89         if client_response.history:
     90             response.history = (
---> 91                 *[await cls.from_client_response(r) for r in client_response.history],
     92             )
     93         return response

c:\users\james\appdata\local\programs\python\python38\lib\site-packages\aiohttp_client_cache\response.py in from_client_response(cls, client_response, expires)
     83         # The encoding may be unset even if the response has been read
     84         try:
---> 85             response.encoding = client_response.get_encoding()
     86         except RuntimeError:
     87             pass

c:\users\james\appdata\local\programs\python\python38\lib\site-packages\aiohttp\client_reqrep.py in get_encoding(self)
    997                 encoding = 'utf-8'
    998             else:
--> 999                 encoding = chardet.detect(self._body)['encoding']
   1000         if not encoding:
   1001             encoding = 'utf-8'

c:\users\james\appdata\local\programs\python\python38\lib\site-packages\cchardet\__init__.py in detect(msg)
     13         }
     14     """
---> 15     encoding, confidence = _cchardet.detect_with_confidence(msg)
     16     if isinstance(encoding, bytes):
     17         encoding = encoding.decode()

src\cchardet\_cchardet.pyx in cchardet._cchardet.detect_with_confidence()

TypeError: object of type 'NoneType' has no len()
```

### Expected behavior
Successfully retrieve the page

### Steps to reproduce the behavior
Covered prior

### Workarounds
Patching the lib to check if _body is set if not return utf8. Not a good solution, really.

### Environment
- aiohttp-client-cache version: 0.4.1
- Python version: 3.8.5
- Platform: win10

**Comment by JWCook** [comment]
Thanks for the detailed bug report! I'll get this fixed soon.

**Comment by JWCook** [comment]
This is now fixed in `0.4.2`!
